src/processHistorian.py

```diff
@@ -284,22 +287,33 @@ def exit(self):
         """
         print("Exiting the ProcessHistorian...")
         print("Waiting for all worker threads to finish...")
-        self._opcua_client.unsubscribe_all()
         # Tell all threads they should stop
-        if hasattr(self, "__push_thread"):
+        if self.__push_thread_obj:
             self.__push_thread_obj.should_exit()
-            self.__push_thread.join()
         for work_obj in self.__opc_thread_objs:
             work_obj.should_exit()
         # Wait for them to be really stopped
+        if self.__push_thread:
+            self.__push_thread.join()
         for thread in self.__opc_threads:
             thread.join()
         print("Disconnecting from OPC UA Server...")
+        self._opcua_client.unsubscribe_all()
         self._opcua_client.disconnect()
         # One last push of the values
-        print("Push remaining values from buffer...")
-        self._buffer.write_points()
-        print("Done! Goodbye!")
+        print("Push remaining values (if any) from buffer...")
+        status = self._buffer.write_points()
+        while status:
+            print("Buffer cannot be sent.")
+            choice = input("Try again?  (Y/n): ")
+            while choice.lower() != "n" and choice.lower() != "y":
+                choice = input("Wrong input! " +
+                               "Try to push buffer again? (Y/n): ")
```

Comment:
That is not true. Big Y means that when you only press return that "yes" is the assumed default option, so a retry should happen.

Reply:
besides "invalid" might be better wording than "wrong"